### ISSUE TITLE
/api returns a 404 error

### ISSUE THREAD

**Issue by ffknob** [ckan/ckan#4616]
### CKAN Version if known (or site URL)
2.7

### Please describe the expected behaviour
When GETting the endpoint /api, it should respond something like:
`{"version": 1}`

### Please describe the actual behaviour
It responds with a 404 error.
But request to, for example` /api/3/action/package_show?id=my-dataset` work just fine.

### What steps can be taken to reproduce the issue? 
I don't know exactly how to reproduce this error. We have developed a custom theme that we include in our production.ini as a plugin, but the error occurs even when we take it off.

Here are some log from the both requests do `/api[/3/action/...]`:

```
GET /api

2019-01-10 13:47:32,147 DEBUG [ckan.config.middleware.pylons_app] Pylons route match: {'url': u'api', 'action': u'view', 'controller': u'template'} Origin: core
2019-01-10 13:47:32,147 DEBUG [ckan.config.middleware] Route support answers for GET /api: [(False, 'flask_app'), (True, 'pylons_app', 'core')]
2019-01-10 13:47:32,147 DEBUG [ckan.config.middleware] Serving request via pylons_app app
2019-01-10 13:47:32,227 INFO  [ckan.lib.base]  /api render time 0.008 seconds
2019-01-10 13:47:32,301 INFO  [ckan.lib.base]  /api render time 0.008 seconds


GET /api/3/action/package_show?id=solicitacoes-de-informacao-2018

2019-01-10 13:48:16,403 DEBUG [ckan.config.middleware.pylons_app] Pylons route match: {'action': u'action', 'controller': u'api', 'ver': u'/3', 'logic_function': u'package_show'} Origin: core
2019-01-10 13:48:16,403 DEBUG [ckan.config.middleware] Route support answers for GET /api/3/action/package_show: [(False, 'flask_app'), (True, 'pylons_app', 'core')]
2019-01-10 13:48:16,404 DEBUG [ckan.config.middleware] Serving request via pylons_app app
2019-01-10 13:48:16,556 DEBUG [ckan.logic] check access OK - site_read user=opendata
2019-01-10 13:48:16,560 DEBUG [ckan.lib.base] Retrieving request params: UnicodeMultiDict([('id', u'solicitacoes-de-informacao-2018')])
2019-01-10 13:48:16,560 DEBUG [ckan.lib.base] Retrieving request POST: UnicodeMultiDict([])
2019-01-10 13:48:16,560 DEBUG [ckan.lib.base] Retrieving request GET: UnicodeMultiDict([('id', u'solicitacoes-de-informacao-2018')])
2019-01-10 13:48:16,571 DEBUG [ckan.logic] check access OK - package_show user=opendata
2019-01-10 13:48:16,571 DEBUG [ckan.lib.search.query] Package query: {'q': u'name:"solicitacoes-de-informacao-2018" OR id:"solicitacoes-de-informacao-2018"', 'fq': 'site_id:"tcers"', 'rows': 1, 'wt': 'json'}
2019-01-10 13:48:17,157 INFO  [ckan.lib.base]  /api/3/action/package_show render time 0.600 seconds


```

**Comment by ffknob** [comment]
From `ckan/config/middleware/pylons_app.py:can_handle_request(self, environ)`:

```
        NOTE: There is currently a catch all route for GET requests to
        point arbitrary urls to templates with the same name:

            map.connect('/*url', controller='template', action='view')
```

There's no `ckan/templates/api.py`, but when I create one, it uses this file. The point is, I think it should redirect to `ckan/views/api.py`, but this file is never used (it doesn't even get a .pyc created).

Our installation comes from the `http://packaging.ckan.org/python-ckan_2.7-trusty_amd64.deb` package.

**Comment by ffknob** [comment]
Ok, so I've found out two places that respond for the '/api' endpoint:

1. `ckan/controllers/api.py:get_api()`
2. `ckan/views/api.py:get_api()`

I've managed to get the first one called by expliciting adding in `config/routing.py` the following route:

```
map.connect('/api', controller='api', action='get_api', conditions=GET, ver='/1')
```

But...

1. I have no idea when/if the (2) should have being called or not...
2. And also no clue why the `/api` was not registered in our installation... (I've printed out the `config['routes.map']` in `config/middleware/pylons_app.py` and it's not there... I think).
